elf/arch-loongarch.cc

```diff
@@ -346,7 +346,16 @@ void InputSection<E>::apply_reloc_alloc(Context<E> &ctx, u8 *base) {
         write_k12(loc, S + A);
       break;
     case R_LARCH_PCALA_HI20:
-      write_j20(loc, hi20(S + A, P));
+      if (removed_bytes == 4) {
+	// pcalau12i/addi.d has been relaxed to pcaddi
+        i64 rd = bits(*(ul32 *)(contents.data() + rel.r_offset), 4, 0);
+        *(ul32 *)loc = rd | 0x18000000;
+        write_j20(loc, (S + A - P) >> 2);
+	// skip the next R_LARCH_RELAX, R_LARCH_PCALA_LO12, R_LARCH_RELAX
+	i += 3;
```

Comment:
Do not rewrite the instruction if it is removed. Instead, do this in LO12. See how we handle R_RISCV_TPREL_LO12_I in arch-riscv.cc, for example.

Reply:
> Do not rewrite the instruction if it is removed. Instead, do this in LO12. See how we handle R_RISCV_TPREL_LO12_I in arch-riscv.cc, for example.

Sorry for this oversight.